Production/plugins/TauTupleProducer.cc

```diff
@@ -289,28 +233,28 @@ class TauTupleProducer : public edm::EDAnalyzer {
 
         tauTuple().met_pt = METs->at(0).pt();
         tauTuple().met_phi = METs->at(0).phi();
-	tauTuple().metcov_00 = METs->at(0).getSignificanceMatrix()[0][0];
+        tauTuple().metcov_00 = METs->at(0).getSignificanceMatrix()[0][0];
         tauTuple().metcov_01 = METs->at(0).getSignificanceMatrix()[0][1];
         tauTuple().metcov_11 = METs->at(0).getSignificanceMatrix()[1][1];
-	tauTuple().puppimet_pt = puppiMETs->at(0).pt();
+        tauTuple().puppimet_pt = puppiMETs->at(0).pt();
         tauTuple().puppimet_phi = puppiMETs->at(0).phi();
         tauTuple().puppimetcov_00 = puppiMETs->at(0).getSignificanceMatrix()[0][0];
         tauTuple().puppimetcov_01 = puppiMETs->at(0).getSignificanceMatrix()[0][1];
         tauTuple().puppimetcov_11 = puppiMETs->at(0).getSignificanceMatrix()[1][1];
-	tauTuple().deepmet_pt = deepMETs->at(0).pt();
-	tauTuple().deepmet_phi = deepMETs->at(0).phi();
+        tauTuple().deepmet_pt = deepMETs->at(0).pt();
+        tauTuple().deepmet_phi = deepMETs->at(0).phi();
         tauTuple().genmet_pt = genMETs->at(0).pt();
         tauTuple().genmet_phi = genMETs->at(0).phi();
-	tauTuple().tauSpinnerWTEven = (*tauSpinnerWTEven);
-        tauTuple().tauSpinnerWTOdd = (*tauSpinnerWTOdd);
-        tauTuple().tauSpinnerWTMM = (*tauSpinnerWTMM);
+        tauTuple().tauSpinnerWTEven = tauSpinnerWTEven.isValid() ? *tauSpinnerWTEven : -1.f;
+        tauTuple().tauSpinnerWTOdd = tauSpinnerWTOdd.isValid() ? *tauSpinnerWTOdd : -1.f;
+        tauTuple().tauSpinnerWTMM = tauSpinnerWTMM.isValid() ? *tauSpinnerWTMM : -1.f;
```

Comment:
Do I understand correctly that the workaround for crashes is this `isValid` check? And for ttbar mostly it isn't valid?

Reply:
`isValid` is method of Handle. If handle is not valid it means that the corresponding producer is not run. Which is the case if `runTauSpinner=False`.